client-runtime/src/main/java/com/microsoft/rest/v2/http/HttpResponse.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.rest.v2.http;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * This class contains all of the details necessary for reacting to a HTTP response from a
+ * HttpResponse.
+ */
+public interface HttpResponse {
+    /**
+     * Get whether or not this response object has a body.
+     * @return Whether or not this response object has a body.
+     */
+    boolean hasBody();
+
+    /**
+     * Get this response object's body as an InputStream. If this response object doesn't have a
+     * body, then null will be returned.
+     * @return This response object's body as an InputStream. If this response object doesn't have a
+     * body, then null will be returned.
+     */
+    InputStream getBodyAsInputStream();
```

Comment:
I want to make sure its clear that it's the body that is being returned as an InputStream, byte[], or String, so can we settle on just getting rid of "get" and leaving it as "bodyAsInputStream()"?